pkg/commands/chgm/chgm.go

```diff
@@ -0,0 +1,137 @@
+package chgm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/openshift/configuration-anomaly-detection/pkg/aws"
+	"github.com/openshift/configuration-anomaly-detection/pkg/ocm"
+
+	"github.com/aws/aws-sdk-go/service/cloudtrail"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	v1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+)
+
+// ChgmImpl should have all of the functions that IsRunningService is implementing
+type ChgmImpl struct {
+	AwsClient aws.Client
+	OcmClient ocm.Client
+}
+
+// ChgmService will wrap all the required commands IsRunning needs to run it's operations
+//go:generate mockgen -source $GOFILE -destination ./mock/interfaces.go -package mock
+type ChgmService interface {
+	// AWS
+	ListStopInstancesEvents() ([]*cloudtrail.Event, error)
+	ListRunningInstances(infraId string) ([]*ec2.Instance, error)
+	ListStoppedInstances(infraId string) ([]*ec2.Instance, error)
+	// OCM
+	GetClusterDeployment(clusterID string) (*hivev1.ClusterDeployment, error)
+	GetClusterInfo(identifier string) (*v1.Cluster, error)
+}
+
+// Chgm is an implementation of the Interface, and adds functionality above it
+// to craete Chgm you can use a mock, or fill it with IsRunningImpl
+type Chgm struct {
+	ChgmService
+}
+
+// isUserAllowedToStop will verify if a user is allowed to stop instances.
+// as this is the thing that might change the most it is at the top.
+// additionally it is private as I don't see anyone using this outside of AreAllInstancesRunning
+func isUserAllowedToStop(username, infraID string) bool {
+	//TODO: what is the best ordering for this? (from the most common to the most rare)
+
+	// pulled by:
+	// 1. logging (via osdctl account cli) into an aws cluster
+	// 2. running the command "aws iam list-users --query 'Users[?starts_with(UserName,`<INFRA_ID>`)].UserName'"
+	// 3. trimming the infra id from the front and the uuid from the back
+	operatorIamNames := []string{
+		"cloud-credential-operator-iam-ro",
+		"cloud-ingress-operator-credentials-aw",
+		"instance",
+		"managed-velero-operator-iam-credentia",
+		"openshift-image-registry",
+		"openshift-ingress",
+		"openshift-machine-api-aws",
```

Comment:
When we are using the actual defined operator IAM here and not limit it to a generic regex like 
`case strings.HasPrefix(username, infraID)`
, it may make sense to reduce the list to the ones we know are actually allowed to do here